code/game/gamemodes/vampire/vampire.dm

```diff
@@ -494,6 +525,75 @@ You are weak to holy things and starlight. Don't go into space and avoid the Cha
 				IgniteMob()
 	adjustFireLoss(3)
 
+/mob/living/carbon/human/proc/handle_vampire_smite()
+	var/smitetemp = 0
+	var/vampcoat = istype(wear_suit, /obj/item/clothing/suit/storage/draculacoat) //coat reduces smiting
+	if(check_holy()) //if you're on a holy tile get ready for pain
+		smitetemp += (vampcoat ? 1 : 5)
+		if(prob(35))
+			src << "<span class='danger'>This ground is blessed. Get away, or splatter it with blood to make it safe for you.</span>"
+
+	if(istype(loc.loc, /area/chapel)) //stay out of the chapel unless you want to turn into a pile of ashes
```

Comment:
don't use ```loc.loc```, if you want an area use ```get_area()```
```loc.loc``` is not guaranteed to be an area
eg it they're in a locker, it'll be a turf